### ISSUE TITLE
Having trouble with additional_copy_path and additional_copy_roles_path

### ISSUE THREAD

**Issue by jimmynapkins** [neillturner/kitchen-ansible#248]
In my .kitchen.yml file I have tried to specify the following

```
roles_path: ../../roles
additional_copy_roles_path: [ ../../roles-galaxy ]
```
Looking at my vagrant VM I see the roles-galaxy folder is copied over but when ansible runs I get the following error:

```ERROR! the role 'iptables' was not found in /tmp/kitchen/roles:/tmp/kitchen/roles:/tmp/kitchen```

the iptables role lives in that roles-galaxy folder. I also tried the following in a custom ansible.cfg with the ansible_cfg_path option:

roles_path = roles-galaxy:roles

Any idea what I am missing here? Any help would be appreciated. Thanks!

**Comment by jimmynapkins** [comment]
Also I am using - kitchen-ansible (0.47.0, 0.45.7)

**Comment by jimmynapkins** [comment]
Okay I noticed I misspelled `additional_copy_roles_path` so I changed it to `additional_copy_role_path` and now it shows up in the error as such:

```ERROR! the role 'iptables' was not found in /tmp/kitchen/roles:/tmp/kitchen/roles:/Users/jimmy/Projects/sauce-ansible/roles-galaxy:/tmp/kitchen```

Shouldn't that be `/tmp/kitchen/roles-galaxy` and not the path on my local machine?

**Comment by jimmynapkins** [comment]
Destroyed the vm and started fresh. Now I get this error:

```       ERROR! the role 'iptables' was not found in /tmp/kitchen/roles:/tmp/kitchen/roles:/tmp/kitchen/roles-galaxy:/tmp/kitchen

       The error appears to have been in '/tmp/kitchen/nginx.yml': line 24, column 7, but may
       be elsewhere in the file depending on the exact syntax problem.

       The offending line appears to be:

           - { role: network_interface, tags: networking }
           - { role: iptables, tags: iptables }
             ^ here```

**Comment by neillturner** [comment]
I don't believe setting
 roles_path: ../../roles
with this sort of relative path will work. 
We have recently added support for this sort of relative path (using the ruby File.expand_path) for some of the parameters like additional_copy_roles_path

**Comment by mikef-nl** [comment]
Could you give us a tree of what the folder structure looks like too?

**Comment by jimmynapkins** [comment]
Hey all, 

roles_path: ../../roles seems to be working. If I change the path to anything else then the playbook fails because it cannot load the first role. However if I change it back to ../../roles it loads every role in that directory until it gets to the iptables role which lives in ../../roles-galaxy.

Here's an edited tree:

```
company-ansible-repo/
├── roles
│   ├── nginx
├── roles-galaxy
│   ├── iptables
├── tests
│   ├── inspec
│   ├── inventory_tests
│   └── nginx <----- .kitchen.yml here (so ../../roles and ../../roles-galaxy)
│       └── config
```

If you need a more detailed explanation let me know. And thanks for the help!

**Comment by jimmynapkins** [comment]
So to make sure I didn't have any lingering weirdness on the VM I was testing against I completely destroyed the VM and its files, then I ran kitchen create, then kitchen converge. 

Verified my .kitchen.yml contains:

```  
roles_path: ../../roles/
additional_copy_role_path: ../../roles-galaxy
```

Then the following error:

```
ERROR! the role 'iptables' was not found in /tmp/kitchen/roles:/tmp/kitchen/roles:/Users/jimmy/Projects/company-ansible/roles-galaxy:/tmp/kitchen
```

You'll notice in the error above everything references /tmp/kitchen on the VM except for the roles-galaxy directory which shows the absolute path on my local machine:

`/Users/jimmy/Projects/company-ansible/roles-galaxy`

If I ssh to that vagrant VM and list the files in `/tmp/kitchen` I see the following:

```
root@nginx-vagrant:~# ls -lah /tmp/kitchen/
total 64K
drwxrwxr-x  12 ubuntu ubuntu 4.0K Jun 21 18:07 .
drwxrwxrwt   8 root   root   4.0K Jun 21 18:08 ..
-rw-r--r--   1 ubuntu ubuntu   22 Jun 21 18:07 ansible.cfg
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 callback_plugins
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 filter_plugins
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 group_vars
-rw-r--r--   1 ubuntu ubuntu   55 Jun 21 18:07 hosts
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 host_vars
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 library
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 lookup_plugins
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 modules
-rw-r--r--   1 ubuntu ubuntu 2.2K Jun 21 18:07 nginx.yml
drwxr-xr-x 159 ubuntu ubuntu 4.0K Jun 21 18:08 roles
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 spec
drwxr-xr-x   2 ubuntu ubuntu 4.0K Jun 21 18:07 ssh_private_keys
-rw-r--r--   1 ubuntu ubuntu   33 Jun 21 18:07 vault
```

Also ran a find to search for the roles-galaxy on the VM and it is nowhere to be found. 

I hope this helps. Thanks!

**Comment by neillturner** [comment]
All the roles get copies to the /tmp/kitchen/roles area on the VM. kitchen ansibles doesn't copy everything to the same place on the VM as you have it on your repo. 
logon to the vm and the iptables role should be in the /tmp/kitchen/role directory. 
assuming the iptables role is in 
 /Users/jimmy/Projects/company-ansible/roles-galaxy
then i would specify:
 roles_path: /Users/jimmy/Projects/company-ansible/roles-galaxy
and see if it gets copied to /tmp/kitchen/roles
if not then we need to put some puts messages in the code to see why 
you should run all your kitchen commands suffixed with -l debug 
get the debugging messages

**Comment by mikef-nl** [comment]
I see the issue, it is indeed adding the full path to the roles config. I will create a fix and pull request.

**Comment by jimmynapkins** [comment]
Awesome. Thanks!

**Comment by mikef-nl** [comment]
Pull request created https://github.com/neillturner/kitchen-ansible/pull/249

**Comment by jimmynapkins** [comment]
Just confirmed that this resolves the issue. Thanks again!

**Comment by neillturner** [comment]
closed fixed in version 0.47.1
